vignettes/function-flow.Rmd

```diff
@@ -0,0 +1,161 @@
+help with development and understanding of the code. They are not hard
+rules.
+
+### Naming
+
+-   First word is an action verb, later words are objects or conditions.
+-   Exclusion criteria are prefixed with `exclude_`.
+-   Inclusion criteria are prefixed with `include_`.
+-   Helpers that get or extract a condition ("pregnancy" or "date of
+    visit") are prefixed with `get_`.
+-   Helpers that drop or keep a specific condition are prefixed with
+    `drop_` or `keep_` ("first visit date to maternal care for pregnancy
+    after 40 weeks"). These types of helpers likely are contained in the
+    `get_` functions.
+
+### Input and output
+
+-   Few arguments, with one or two core required argument.
+-   `include_` functions take a register as the first argument.
+    -   One input register database at a time.
+-   `exclude_` functions can take a register as the first argument or
+    take the output from an `include_` function.
+-   Second argument can be an output data from another function.
+
+## Function flow
+
+The OSDC algorithm - and thereby, the `osdc` package - contains one main
+function that will classify individuals into those with either type 1 or
+type 2 diabetes using the Danish registers: `classify_diabetes()`. This
+function classifies those with diabetes (type 1 or 2) based on the
+Danish registers described in the `vignette("design")`. All data sources
+are used as input for this function. The specific inclusion and
+exclusion details are also described in the `vignette("design")`.
+
+This results in the functionality flow for classifying diabetes status
+seen below. All functions take a `data.frame` type object as input and
+outputs the same type of object as the input object (a `data.frame`
+type). For instance, if the input is a `data.table` object, the output
+will also be a `data.table`.
+
+![Flow of functions, as well as their required input registers, for
+classifying diabetes status using the `osdc` package. Light blue and
+orange boxes represent filtering functions (inclusion and exclusion
+events, respectively).](images/function-flow.png)
+
+## Inclusion events
+
+### HbA1c tests above 48 mmol/mol
+
+The function `include_hba1c()` uses `lab_forsker` as the input data to
+extract all events of tests above 48 mmol/mol.
+
+<!-- TODO: Add details on how this filtering should be done -->
+
+### Hospital diagnosis of diabetes
+
+The function `include_diabetes_diagnoses()` uses the hospital contacts
+from LPR2 and 3 to include all dates of diabetes diagnoses. Diabetes
+diagnoses from both ICD 8 and ICD 10 are included.
+
+This function contains two helper functions:
+
+-   `keep_diabetes_icd10()`
+-   `keep_diabetes_icd8()`
+
+<!-- TODO: Add details on how this filtering should be done, e.g., diagnosis codes -->
+
+<!-- TODO: Which specific ICD 8 and 10 codes are included? -->
+
+### Diabetes-specific podiatrist services
+
+The function `include_podiatrist_services()` uses `sysi` or `sssy` as
```

Comment:
Should we include what the abbreviations sysi and sssy mean here? - same goes for e.g., lmdb in the next section

Reply:
Not really, this document isn't about describing that stuff, that's what the other documents are for. This is only for the function flow.